git_machete/cmd.py

```diff
@@ -2486,47 +1820,41 @@ def get_relation_to_remote_counterpart(cli_ctxt: CommandLineContext, b: str, rb:
+        b_t = git_operations.committer_unix_timestamp_by_revision(cli_ctxt, b, "refs/heads/")
+        rb_t = git_operations.committer_unix_timestamp_by_revision(cli_ctxt, rb, "refs/remotes/")
         return DIVERGED_FROM_AND_OLDER_THAN_REMOTE if b_t < rb_t else DIVERGED_FROM_AND_NEWER_THAN_REMOTE
 
 
 def get_strict_remote_sync_status(cli_ctxt: CommandLineContext, b: str) -> Tuple[int, Optional[str]]:
-    if not remotes(cli_ctxt):
+    if not git_operations.remotes(cli_ctxt):
         return NO_REMOTES, None
-    rb = strict_counterpart_for_fetching_of_branch(cli_ctxt, b)
+    rb = git_operations.strict_counterpart_for_fetching_of_branch(cli_ctxt, b)
     if not rb:
         return UNTRACKED, None
-    return get_relation_to_remote_counterpart(cli_ctxt, b, rb), strict_remote_for_fetching_of_branch(cli_ctxt, b)
+    return get_relation_to_remote_counterpart(cli_ctxt, b, rb), git_operations.strict_remote_for_fetching_of_branch(cli_ctxt, b)
 
 
 def get_combined_remote_sync_status(cli_ctxt: CommandLineContext, b: str) -> Tuple[int, Optional[str]]:
-    if not remotes(cli_ctxt):
+    if not git_operations.remotes(cli_ctxt):
         return NO_REMOTES, None
-    rb = combined_counterpart_for_fetching_of_branch(cli_ctxt, b)
+    rb = git_operations.combined_counterpart_for_fetching_of_branch(cli_ctxt, b)
     if not rb:
         return UNTRACKED, None
-    return get_relation_to_remote_counterpart(cli_ctxt, b, rb), combined_remote_for_fetching_of_branch(cli_ctxt, b)
+    return get_relation_to_remote_counterpart(cli_ctxt, b, rb), git_operations.combined_remote_for_fetching_of_branch(cli_ctxt, b)
 
 
 # Reflog magic
 
 
-REFLOG_ENTRY = Tuple[str, str]
-
-reflogs_cached: Optional[Dict[str, Optional[List[REFLOG_ENTRY]]]] = None
-
-
 def load_all_reflogs(cli_ctxt: CommandLineContext) -> None:
-    global reflogs_cached
     # %gd - reflog selector (refname@{num})
     # %H - full hash
     # %gs - reflog subject
-    all_branches = [f"refs/heads/{b}" for b in local_branches(cli_ctxt)] + \
-                   [f"refs/remotes/{combined_counterpart_for_fetching_of_branch(cli_ctxt, b)}" for b in local_branches(cli_ctxt) if combined_counterpart_for_fetching_of_branch(cli_ctxt, b)]
+    all_branches = [f"refs/heads/{b}" for b in git_operations.local_branches(cli_ctxt)] + \
+                   [f"refs/remotes/{git_operations.combined_counterpart_for_fetching_of_branch(cli_ctxt, b)}" for b in git_operations.local_branches(cli_ctxt) if git_operations.combined_counterpart_for_fetching_of_branch(cli_ctxt, b)]
     # The trailing '--' is necessary to avoid ambiguity in case there is a file called just exactly like one of the branches.
-    entries = non_empty_lines(popen_git(cli_ctxt, "reflog", "show", "--format=%gD\t%H\t%gs", *(all_branches + ["--"])))
-    reflogs_cached = {}
+    entries = utils.non_empty_lines(git_operations.popen_git(cli_ctxt, "reflog", "show", "--format=%gD\t%H\t%gs", *(all_branches + ["--"])))
+    GitContext.reflogs_cached = {}
```

Comment:
Hmmm that's a code smell &mdash; things like a cache should be an implementation detail of `GitContext`, **not** exposed to the outside of `GitContext`.

Other than making fields like `reflogs_cached` private (which can be done in one go on a separate PR), I'd consider moving this entire `load_all_reflogs` method directly to `GitContext` &mdash; there's nothing inherently Machete-specific here AFAICS.